javaslang/src/test/java/javaslang/collection/QuickSort.java

```diff
@@ -0,0 +1,33 @@
+/*     / \____  _    _  ____   ______  / \ ____  __    _______
+ *    /  /    \/ \  / \/    \ /  /\__\/  //    \/  \  //  /\__\   JΛVΛSLΛNG
+ *  _/  /  /\  \  \/  /  /\  \\__\\  \  //  /\  \ /\\/ \ /__\ \   Copyright 2014-2017 Javaslang, http://javaslang.io
+ * /___/\_/  \_/\____/\_/  \_/\__\/__/\__\_/  \_//  \__/\_____/   Licensed under the Apache License, Version 2.0
+ */
+package javaslang.collection;
+
+import javaslang.Function1;
+import org.junit.Test;
+
+import static java.lang.Integer.signum;
+import static javaslang.API.List;
+
+public class QuickSort {
+    @Test
+    public void shouldQuickSort() {
+        final List<Integer> values = List(3, 1, 4, 1, 5, 9, 2, 6, 5, 3, 5, 8, 9, 7, 9, 3, 2, 3, 8, 4, 6, 2, 6);
+        assert sort(values).equals(values.sorted());
+        assert sort2(values).equals(values.sorted());
```

Comment:
interesting approach to use Java's assert keyword in unit tests - haven't done that, yet :)